velox/experimental/wave/common/Cuda.cu

```diff
@@ -135,15 +136,7 @@ Device* currentDevice() {
 }
 
 Device* getDevice(int32_t deviceId) {
-  Device* save = nullptr;
-  if (driverInited) {
-    save = currentDevice();
-  }
-  auto* dev = setDriverDevice(deviceId);
-  if (save) {
-    setDevice(save);
```

Comment:
We are changing the current device to `deviceId` with this change, can we keep the logic to restore it back to it's old value?

Reply:
Make sense! Fixed! I felt that getDevice means to set it as the current device but not always true.